pkg/services/updatechecker/grafana.go

```diff
@@ -57,20 +72,36 @@ func (s *GrafanaService) Run(ctx context.Context) error {
 	return ctx.Err()
 }
 
-func (s *GrafanaService) checkForUpdates() {
-	resp, err := s.httpClient.Get("https://raw.githubusercontent.com/grafana/grafana/main/latest.json")
+func (s *GrafanaService) checkForUpdates(ctx context.Context) {
```

Comment:
I think it may be cleaner to just duplicate code like so, instead of worrying about the `err` in the defer (shadowing, etc.)

```suggestion
func (s *GrafanaService) checkForUpdates(ctx context.Context) {
	var err error
	ctx, span := s.tracer.Start(ctx, "updatechecker.GrafanaService.checkForUpdates")
	defer span.End()

	ctxLogger := s.log.FromContext(ctx)
	ctxLogger.Debug("Checking for updates")
	resp, err := s.httpClient.Get(ctx, "https://raw.githubusercontent.com/grafana/grafana/main/latest.json")
	if err != nil {
		span.SetStatus(codes.Error, "update check failed: "+err.Error())
		span.RecordError(err)
		ctxLogger.Debug("Failed to get latest.json repo from github.com", "error", err)
		return
	}
	defer func() {
		if err := resp.Body.Close(); err != nil {
			ctxLogger.Warn("Failed to close response body", "err", err)
		}
	}()
	body, err := io.ReadAll(resp.Body)
	if err != nil {
		span.SetStatus(codes.Error, "update check failed: "+err.Error())
		span.RecordError(err)
		ctxLogger.Debug("Update check failed, reading response from github.com", "error", err)
		return
	}

	type latestJSON struct {
		Stable  string `json:"stable"`
		Testing string `json:"testing"`
	}
	var latest latestJSON
	err = json.Unmarshal(body, &latest)
	if err != nil {
		span.SetStatus(codes.Error, "update check failed: "+err.Error())
		span.RecordError(err)
		ctxLogger.Debug("Failed to unmarshal latest.json", "error", err)
		return
	}

	ctxLogger.Debug("Update check succeeded")

	s.mutex.Lock()
	defer s.mutex.Unlock()
	if strings.Contains(s.grafanaVersion, "-") {
		s.latestVersion = latest.Testing
		s.hasUpdate = !strings.HasPrefix(s.grafanaVersion, latest.Testing)
	} else {
		s.latestVersion = latest.Stable
		s.hasUpdate = latest.Stable != s.grafanaVersion
	}

	currVersion, err1 := version.NewVersion(s.grafanaVersion)
	latestVersion, err2 := version.NewVersion(s.latestVersion)
	if err1 == nil && err2 == nil {
		s.hasUpdate = currVersion.LessThan(latestVersion)
	}
}
```
```